app/api/helpers/ticketing.py

```diff
@@ -151,6 +159,14 @@ def charge_paypal_order_payment(order, token_id):
 
                 return True, order
             else:
+                # payment failed hence expire the order
+                order.status = 'expired'
```

Comment:
Is it immediately expired? Or is there a provision to wait for a while and retry? I guess retrying creates a different order, right?